packages/sdk/src/TemplateProvider.ts

```diff
@@ -6,51 +6,91 @@ import * as fs from 'fs'
 import { CLDebug } from './CLDebug'
+import { TemplateEngine, ActivityFactory } from 'botbuilder-lg'
 
 /**
- * Provider for rendering templates
+ * Provider for rendering LG templates.
  */
 export class TemplateProvider {
+
     private static hasSubmitError = false
 
+    public static LGTemplateDirectory(): string | null {
+        const cwd = process.cwd()
+        const lgDirectoryName = "lg"
+        // TODO - make this configurable.
+        let templateDirectory = path.join(cwd, `./${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        // Try up a directory or two as could be in /lib or /dist folder depending on deployment
+        templateDirectory = path.join(cwd, `../${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        templateDirectory = path.join(cwd, `../../${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        return null
+    }
+
     // TODO: Decouple template renderer from types from Action classes
     // E.g. use generic key,value object instead of RenderedActionArgument
-    public static async RenderTemplate(templateName: string, templateArguments: RenderedActionArgument[]): Promise<any | null> {
-        let template = this.GetTemplate(templateName)
-        if (template === null) {
+    public static async RenderTemplate(templateName: string, templateArguments: RenderedActionArgument[], entityDisplayValues: Map<string, string>): Promise<any | null> {
+        const templateDirectory = this.LGTemplateDirectory()
+        if (templateDirectory === null) {
             return null
         }
 
-        let templateString = JSON.stringify(template)
-        let argumentNames = this.GetArgumentNames(templateString)
-
-        // Substitute argument values
-        for (let argumentName of argumentNames) {
-            let renderedActionArgument = templateArguments.find(a => a.parameter == argumentName)
-            if (renderedActionArgument) {
-                templateString = templateString.replace(new RegExp(`{{${argumentName}}}`, 'g'), renderedActionArgument.value ?? '')
-            }
+        const renderingParameters = {}
+        // Copy parameters defined in the action to our rendering parameters.
+        for (const templateArgument of templateArguments) {
+            renderingParameters[templateArgument.parameter] = templateArgument.value
         }
-        templateString = this.RemoveEmptyArguments(templateString)
-        return JSON.parse(templateString)
+        // Copy entities in memory to our rendering parameters.
+        entityDisplayValues.forEach((value: string, key: string) => { renderingParameters[key] = value })
+
+        // Load the template.  Currently, we assume that each lg file only has one template.
+        const lgFilename = templateDirectory + "//" + templateName + ".lg"
```

Comment:
should construct paths using `path` library. `path.join` handles OS differences

Reply:
Missed this one when going through the code port, thanks.